pkg/binder/cache/commonstores/node_store/node_store.go

```diff
@@ -0,0 +1,519 @@
+	}
+	s.addNodeImageStates(node, nodeInfo)
+	nodeInfo.SetNodeInSchedulerPartition(nodeutil.NodeOfThisScheduler(node.Annotations, s.handler.ComponentName()))
+	err := nodeInfo.SetNode(node)
+	s.Add(node.GetName(), nodeInfo)
+	return err
+}
+
+func (s *NodeStore) AddNMNode(nmNode *nodev1alpha1.NMNode) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(nmNode.GetName()); obj != nil {
+		nodeInfo = obj
+		s.Deleted.Delete(nmNode.GetName())
+		s.Delete(nmNode.GetName(), nodeInfo)
+	} else {
+		nodeInfo = framework.NewNodeInfo()
+	}
+	nodeInfo.SetNMNodeInSchedulerPartition(nodeutil.NodeOfThisScheduler(nmNode.Annotations, s.handler.ComponentName()))
+	err := nodeInfo.SetNMNode(nmNode)
+	s.Add(nmNode.GetName(), nodeInfo)
+	return err
+}
+
+func (s *NodeStore) AddCNR(cnr *katalystv1alpha1.CustomNodeResource) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(cnr.Name); obj != nil {
+		nodeInfo = obj
+	} else {
+		nodeInfo = framework.NewNodeInfo()
+	}
+	err := nodeInfo.SetCNR(cnr)
+	s.Set(cnr.Name, nodeInfo)
+	return err
+}
+
+func (s *NodeStore) UpdateNode(oldNode, newNode *v1.Node) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(newNode.GetName()); obj != nil {
+		nodeInfo = obj
+		s.removeNodeImageStates(nodeInfo.GetNode())
+		s.Deleted.Delete(newNode.GetName())
+		s.Delete(newNode.GetName(), nodeInfo)
+	} else {
+		nodeInfo = framework.NewNodeInfo()
+	}
+	s.addNodeImageStates(newNode, nodeInfo)
+	nodeInfo.SetNodeInSchedulerPartition(nodeutil.NodeOfThisScheduler(newNode.Annotations, s.handler.ComponentName()))
+	err := nodeInfo.SetNode(newNode)
+	s.Add(newNode.GetName(), nodeInfo)
+	return err
+}
+
+func (s *NodeStore) UpdateNMNode(oldNMNode, newNMNode *nodev1alpha1.NMNode) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(newNMNode.GetName()); obj != nil {
+		nodeInfo = obj
+		s.Deleted.Delete(newNMNode.GetName())
+		s.Delete(newNMNode.GetName(), nodeInfo)
+	} else {
+		nodeInfo = framework.NewNodeInfo()
+	}
+	nodeInfo.SetNMNodeInSchedulerPartition(nodeutil.NodeOfThisScheduler(newNMNode.Annotations, s.handler.ComponentName()))
+	err := nodeInfo.SetNMNode(newNMNode)
+	s.Add(newNMNode.GetName(), nodeInfo)
+	return err
+}
+
+func (s *NodeStore) UpdateCNR(oldCNR, newCNR *katalystv1alpha1.CustomNodeResource) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(newCNR.Name); obj != nil {
+		nodeInfo = obj
+	} else {
+		nodeInfo = framework.NewNodeInfo()
+	}
+	err := nodeInfo.SetCNR(newCNR)
+	s.Set(newCNR.Name, nodeInfo)
+	return err
+}
+
+func (s *NodeStore) DeleteNode(node *v1.Node) error {
+	var nodeInfo framework.NodeInfo
+	if obj := s.Get(node.GetName()); obj != nil {
+		nodeInfo = obj
+	} else {
+		return fmt.Errorf("node %v is not found", node.GetName())
+	}
+	s.Delete(node.GetName(), nodeInfo)
+	nodeInfo.RemoveNode()
+	s.removeNodeImageStates(node)
+
+	if nodeInfo.GetNMNode() != nil {
+		s.Add(node.GetName(), nodeInfo)
+	} else if nodeInfo.NumPods() != 0 || nodeInfo.GetCNR() == nil {
```

Comment:
nit: this should be `nodeInfo.NumPods() != 0 || nodeInfo.GetCNR() != nil`

Reply:
Ah, good catch! Fixed.